File: home/dot_local/share/lazyvim/lib/paths.mjs
import path from "node:path";
import { NODE_VERSION } from "./constants.mjs";

export const isWindows = process.platform === "win32";
export const isLinux = process.platform === "linux";
export const home =
  process.env.CHEZMOI_DESTDIR || process.env.HOME || process.env.USERPROFILE;

if (!home) throw new Error("Unable to determine the target home directory");

export const local = path.join(home, ".local");
export const managedNode = isWindows
  ? path.join(local, "opt", "nvm-windows", "nodejs", "node.exe")
  : path.join(
      local,
      "opt",
      "nvm",
      "versions",
      "node",
      `v${NODE_VERSION}`,
      "bin",
      "node",
    );
export const managedNvim = isWindows
  ? path.join(local, "opt", "nvim", "bin", "nvim.exe")
  : path.join(local, "bin", "nvim");

export function tool(name) {
  return path.join(local, "bin", `${name}${isWindows ? ".exe" : ""}`);
}

File: home/dot_local/share/lazyvim/lua/lazyvim_capabilities/paths.lua
local M = {}

M.home = vim.env.CHEZMOI_DESTDIR or vim.env.HOME or vim.env.USERPROFILE
assert(M.home and M.home ~= "", "unable to determine target home")
M.home = vim.fs.normalize(M.home)
M.local_dir = vim.fs.joinpath(M.home, ".local")

function M.join(...)
	return vim.fs.joinpath(...)
end

function M.read(path)
	local file = assert(io.open(path, "rb"))
	local contents = file:read("*a")
	file:close()
	return contents
end

function M.write(path, contents)
	vim.fn.mkdir(vim.fs.dirname(path), "p")
	local file = assert(io.open(path, "wb"))
	assert(file:write(contents))
	file:close()
end

function M.exists(path)
	return vim.uv.fs_stat(path) ~= nil
end

return M
